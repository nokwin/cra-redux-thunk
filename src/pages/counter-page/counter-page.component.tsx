import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { Helmet } from "react-helmet";

import {
  incrementAsyncAction,
  decrementAsyncAction,
} from "../../redux/counter/counter.thunks";
import { selectCount } from "../../redux/counter/counter.selectors";

export const CounterPage = () => {
  const dispatch = useDispatch();

  const increment = () => {
    dispatch(incrementAsyncAction());
  };
  const decrement = () => {
    dispatch(decrementAsyncAction());
  };
  const count = useSelector(selectCount);

  return (
    <>
      <Helmet>
        <title>Counter</title>
      </Helmet>
      <div>
        <button onClick={increment}>async +</button>
        {count}
        <button onClick={decrement}>async -</button>
      </div>
    </>
  );
};
