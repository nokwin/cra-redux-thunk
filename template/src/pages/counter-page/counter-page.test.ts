import { CounterPage } from "./counter-page.component";

describe("Counter page component", () => {
  it("should render counter page component", () => {
    expect(CounterPage).toMatchSnapshot();
  });
});
