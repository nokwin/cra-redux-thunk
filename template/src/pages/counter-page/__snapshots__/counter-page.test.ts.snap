// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter page component should render counter page component 1`] = `[Function]`;
